subsys/net/openthread/rpc/server/ot_rpc_commissioning_server.c

```diff
@@ -0,0 +1,169 @@
+		pkt_data.lqi = result->mLqi;
+		pkt_data.version = result->mVersion;
+		pkt_data.is_native = result->mIsNative;
+		pkt_data.discover = result->mDiscover;
+		pkt_data.is_joinable = result->mIsJoinable;
+
+		nrf_rpc_encode_buffer(&ctx, &pkt_data, sizeof(pkt_data));
+	} else {
+		nrf_rpc_encode_null(&ctx);
+	}
+
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)context);
+	nrf_rpc_encode_uint(&ctx, callback_slot);
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_THREAD_DISCOVER_CB, &ctx, ot_rpc_decode_void,
+				NULL);
+}
+
+NRF_RPC_CBKPROXY_HANDLER(ot_thread_discover_cb_encoder, ot_thread_discover_cb,
+			 (otActiveScanResult *result, void *context), (result, context));
+
+static void ot_rpc_thread_discover_rpc_handler(const struct nrf_rpc_group *group,
+					       struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	uint32_t scan_channels;
+	uint16_t pan_id;
+	bool joiner;
+	bool enable_eui64_filtering;
+	otHandleActiveScanResult cb;
+	void *cb_ctx;
+	otError error;
+
+	scan_channels = nrf_rpc_decode_uint(ctx);
+	pan_id = nrf_rpc_decode_uint(ctx);
+	joiner = nrf_rpc_decode_bool(ctx);
+	enable_eui64_filtering = nrf_rpc_decode_bool(ctx);
+	cb = (otHandleActiveScanResult)nrf_rpc_decode_callbackd(ctx, ot_thread_discover_cb_encoder);
+	cb_ctx = (void *)nrf_rpc_decode_uint(ctx);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		ot_rpc_report_decoding_error(OT_RPC_CMD_THREAD_DISCOVER, handler_data);
+		return;
+	}
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	error = otThreadDiscover(openthread_get_default_instance(), scan_channels, pan_id, joiner,
+				 enable_eui64_filtering, cb, cb_ctx);
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_thread_discover, OT_RPC_CMD_THREAD_DISCOVER,
+			 ot_rpc_thread_discover_rpc_handler, NULL);
+
+static void ot_rpc_dataset_is_commissioned_rpc_handler(const struct nrf_rpc_group *group,
+						       struct nrf_rpc_cbor_ctx *ctx,
+						       void *handler_data)
+{
+	bool result;
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		ot_rpc_report_decoding_error(OT_RPC_CMD_DATASET_IS_COMMISSIONED, handler_data);
+		return;
+	}
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	result = otDatasetIsCommissioned(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_bool(group, result);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_dataset_is_commissioned,
+			 OT_RPC_CMD_DATASET_IS_COMMISSIONED,
+			 ot_rpc_dataset_is_commissioned_rpc_handler, NULL);
+
+static void ot_rpc_data_set_active_tlvs_rpc_handler(const struct nrf_rpc_group *group,
+						    struct nrf_rpc_cbor_ctx *ctx,
+						    void *handler_data)
+{
+	otOperationalDatasetTlvs dataset;
+	void *data_ptr = &dataset;
+	otError error;
+
+	ot_rpc_decode_dataset_tlvs(group, ctx, &data_ptr);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
```

Comment:
You should also check if `data_ptr` is not nulled out since ot_rpc_decode_dataset_tlvs allows for NULL.

Reply:
My initial assumption was that Open Thread will check it is not NULL for me and return an appropriate error code if it is NULL.
Is it wrong?

Reply:
The thing was that you always passed a valid `&dataset` pointer even if this was not populated. But now you added the `data_ptr` check so this is OK.